engine/flink/util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/AggregateTransformer.scala

```diff
@@ -0,0 +1,127 @@
+package pl.touk.nussknacker.engine.flink.util.transformer.aggregate
+
+import org.apache.flink.api.common.functions.RichMapFunction
+import org.apache.flink.api.common.state.ValueStateDescriptor
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction
+import org.apache.flink.streaming.api.scala._
+import org.apache.flink.streaming.api.windowing.assigners.TumblingProcessingTimeWindows
+import org.apache.flink.streaming.api.windowing.time.Time
+import org.apache.flink.util.Collector
+import pl.touk.nussknacker.engine.api.context.ContextTransformation
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.{Context => NkContext, _}
+import pl.touk.nussknacker.engine.flink.api.process._
+import pl.touk.nussknacker.engine.flink.api.state.LatelyEvictableStateFunction
+
+import scala.collection.immutable.TreeMap
+import scala.concurrent.duration.Duration
+
+object AggregateTransformer extends CustomStreamTransformer {
```

Comment:
I wonder if we can think about making those all generic transformers unified and think about them as one is the special case for another. Proposed `AggregateTransformer` looks like most generic one. 
1. One thing that is missing for me is filtering (`transformWhen` in `TransformStateTransformer`). How do you handle the scenario: You want to handle stream of events, based on part of them you want to have aggregation and for other part you want to just read value of state? 
2. If we decide to go this way, we should have the same naming for window time, key be and so on.
3. Mayby those special cases like `TransformStateTransformer`, `PreviousValueTransformer` won't be needed then?

Reply:
1. I think I can add it to this function
2. I'll look at that
3. PreviousValueTransformer is kind of different as it has no notion of time and can be more efficient I think than those based on windows